src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.cs

```diff
@@ -621,6 +621,12 @@ private static void SortContents(byte[] buffer, int start, int end)
 
             Debug.Assert(pos == end);
 
+            // If there is only one thing in the contents, we have no sorting work to do and it's already sorted.
+            if (positions.Count == 1)
```

Comment:
If it's enough of a case that it's worth an early exit, then we should probably rewrite 611-622 to something like

```C#
ReadOnlyMemory<byte> encoded = reader.ReadEncodedValue();

if (!reader.HasData)
{
    return;
}

List<(int, int)> positions = new List<(int, int)>
{
    (pos, encoded.Length),
};

pos += encoded.Length;

// the current loop
```

that way we don't create the List object if we're going to early exit.

Reply:
@bartonjs seems reasonable.